api/src/shop/transactions.py

```diff
@@ -395,6 +406,11 @@ def process_cart(member_id: int, cart: List[CartItem]) -> Tuple[Decimal, List[Tr
             content = TransactionContent(product_id=product_id, count=count, amount=amount)
             contents.append(content)
 
+        if labaccess_in_cart and not (member_has_base_membership or base_membership_in_cart):
+            raise BadRequest(
+                "Could not purchase selected subscription. Please buy the base annual membership before lab access."
+            )
+
```

Comment:
I would prefer to break out this business rule as a function instead, which would make it possible to "conditionalize" easier and see what is "business logic".